### ISSUE TITLE
[Q/Request] Audio handling

### ISSUE THREAD

**Issue by o-l-a-v** [TeamNewPipe/NewPipe#941]
There are some limitations or issues when it comes to how newpipe handles audio imo.

# Downloading Audio
When opening downloaded audio through Newpipe v0.11.2:

1. It opens as a popup. This is an issue because:
Player controls do not appear in notifications like regular audio players
The music stops playing when changing app, depending on app chosen to play it with. Google Play Music in my example

2. It does not load 'real' audio players as a choice to open it with. In my example, I'd like to play the audio with GoneMad Music Player

3. It does not list the actual audio formats. M4A (MP4) and webm are containers, both can hold several different types of audio (and video) formats. Would be better to list container and format on stream(s) inside. Audio would end up being WebM (Vorbis), WebM (Opus), M4A (AAC) etc.
For video, maybe WebM (VP9, Opus), MP4 (AVC, AAC), for instance.

# Opus and mp3 support
## Opus
Even though not all Android versions do support Opus, there are downloadable apps that do. Like GoneMAD Media Player. GMPP adds Opus support for all Android versions it supports, so the hard limit for v5 lollipop is not really the case. Please make it possible to show Opus streams as options when:
A) Streaming
a) Only show on supported devices (>5.0.0)
b) Or add opus decoder to the app as fallback
B. When downloading, all android versions.

## MP3
Patents have expired, implementing MP3 support should be straight forward.

**Comment by Rubbix** [comment]
Still no answer from the developer 😒😒😒

**Comment by theScrabi** [comment]
Mp3 support requires ffmpeg to embedded. I wrote in several other issues why this didn't happen yet and why this is not going to happen. Just search for mp3 support.

1. This is your internal player, it has nothing todo with newpipe.

2. It can only support player that allow streamed music. Your player seem to not support this.

3. This is an actual issue that is right. If you want this to be fixed please open a separate issue for this.

Opus support got already requested and is listrd as an issue as long as i remember.

**Comment by anilgulecha** [comment]
@theScrabi What about a project like https://github.com/Sciss/jump3r ? This would require m4a -> wav (lossless), wav -> mp3 provided by above, so this would not increase the size as much as a full instance of ffmpeg.

**Comment by anilgulecha** [comment]
https://github.com/nwaldispuehl/java-lame seems to be another option.

**Comment by theScrabi** [comment]
There is an app called [video transcoder](https://f-droid.org/app/protect.videoeditor) I think we could create an intent interface to that app. This way ffmpeg is encapsulated in another app, and if someone wanted to have this functionallity he could just install this app like an addon.

**Comment by anilgulecha** [comment]
@theScrabi thanks! I've posted a comment on an issue already tracking intent creation for the video-transcoder app.

A seamless integration with another app would indeed be welcome, and would reduce the maintainable surface area for Newpipe.

**Comment by theScrabi** [comment]
@brarcher what do you say, would you like to cooperate with us on that thing?

**Comment by brarcher** [comment]
I like the idea, @theScrabi. We can discuss the details in https://github.com/brarcher/video-transcoder/issues/42.

**Comment by theScrabi** [comment]
Once this is done we should put a link to your transcoder on our readme/website.

**Comment by opusforlife2** [comment]
Opus is here. Tapping gives me Poweramp as option. Check if done.

**Comment by wb9688** [comment]
1 is intended, 2 is not NewPipe's bug and 3 will maybe be in the new downloader at some point
